server/src/main/java/org/elasticsearch/common/logging/DeprecationLogger.java

```diff
@@ -54,16 +51,21 @@ public static DeprecationLogger getLogger(Class<?> aClass) {
      * the "org.elasticsearch" namespace.
      */
     public static DeprecationLogger getLogger(String name) {
-        return new DeprecationLogger(getDeprecatedLoggerForName(name));
+        return new DeprecationLogger(name);
+    }
+
+    private DeprecationLogger(String parentLoggerName) {
```

Comment:
we always create two loggers when using deprecationLogger. But often it won't be necessary. I wonder what is the performance penalty.

We could create a `CompatibleLogger` that would inherit `DeprecationLogger` and would allow to create two loggers and would have two logging methods `compatibleApiWarning` and `deprecate`.
@pugnascotia WDYT?

Reply:
we could also just use a 
```
        this.compatibleLogger = new LazyInitializable<>(()-> LogManager.getLogger(getLoggerName(parentLoggerName, "compatible")));

```